endpoint.bal

```diff
@@ -0,0 +1,536 @@
+        if (requestHeaders is error) {
+            error err = error(TWITTER_ERROR,
+                              message = "Error occurred while encoding");
+            return err;
+        } else {
+            var httpResponse = check self.twitterClient->post(resourcePath, request);
+            var response = check handleResponse(httpResponse);
+            return convertToStatus(response);
+        }
+    }
+
+    # Delete a Retweet.
+    #
+    # + id - The numerical ID of a status
+    # + trimUser - User object including only numerical ID. Omit parameter to receive the complete user object
+    # + return - If success, returns Status object, else returns error
+    @display {label: "Delete a Retweet"} 
+    remote function deleteRetweet(@display {label: "ReTweet Id to delete"} int id, 
+                                  @display {label: "Trim user or not"} boolean? trimUser = ()) 
+                                  returns @tainted @display {label: "Status"} Status|error {
+        http:Request request = new;
+        string oauthString = getOAuthParameters(self.clientId, self.accessToken);
+
+        string resourcePath = UN_RETWEET_ENDPOINT + id.toString() + JSON;
+        var requestHeaders = createRequestHeaders(request, POST, resourcePath, self.clientId, self.clientSecret,
+            self.accessToken, self.accessTokenSecret, oauthString);
+        if (requestHeaders is error) {
+            error err = error(TWITTER_ERROR,
+                              message = "Error occurred while encoding");
+            return err;
+        } else {
+            var httpResponse = check self.twitterClient->post(resourcePath, request);
+            var response = check handleResponse(httpResponse);
+            return convertToStatus(response);
+        }
+    }
+
+    # Search for Tweets matching a query.
+    #
+    # + queryStr - The query string need to be searched
+    # + searchOptions - Optional parameter which specify the search options
+    # + return - If success, returns Status object, else returns error
+    @display {label: "Search statuses by string"} 
+    remote function search(@display {label: "Query string to search"} string queryStr, 
+                           @display {label: "Optional Search Options"} SearchOptions? searchOptions = ()) 
+                           returns @tainted @display {label: "Array of Status"} Status[]|error {
+        string resourcePath = SEARCH_ENDPOINT;
+        string encodedQueryValue = check url:encode(queryStr, UTF_8);
+        string urlParams = "q=" + encodedQueryValue + AMBERSAND;
+        int? count = searchOptions?.count;
+        string? geocode = searchOptions?.geocode;
+        string oauthString = getOAuthParameters(self.clientId, self.accessToken) + urlParams;
+        if(count is int){
+            oauthString = "count=" + count.toString() + AMBERSAND + oauthString;
+        }
+        if(geocode is string){
+            if (geocode != "") {
```

Comment:
Can't we check both at once `    if ((geocode is string) && (geocode != "")) { `

Reply:
That is better approach. Changed it